cmd/thermal-recorder/main.go

```diff
@@ -322,3 +344,47 @@ func writeInitialFramesToFile(writer *cptv.FileWriter, frames []*lepton3.Frame,
 
 	return nil
 }
+
+func newSnapshot() error {
+	if snapshotPending.Load() == true {
+		//TODO return the pending snapshot error also.
+		return errors.New("snapshot already pending")
+	}
+	log.Println("take new snapshot")
+	snapshotPending.Store(true)
+	defer snapshotPending.Store(false)
+	var err error
+	select {
+	case err = <-snapshotChan:
+	case <-time.After(10 * time.Second):
+		err = errors.New("timeout for snapshot")
+	}
+	return err
+}
+
+func frameToSnapshot(dir string, f *lepton3.Frame) error {
+	g16 := image.NewGray16(image.Rect(0, 0, 160, 120))
+	// Max and min are needed for normalization of the frame
+	var valMax uint16
+	var valMin uint16 = 65535
+	for _, row := range f {
+		for _, val := range row {
+			valMax = uint16(math.Max(float64(valMax), float64(val)))
+			valMin = uint16(math.Min(float64(valMin), float64(val)))
+		}
+	}
+
+	var norm = 65535 / (valMax - valMin)
+	for y, row := range f {
+		for x, val := range row {
+			g16.SetGray16(x, y, color.Gray16{Y: (val - valMin) * norm})
+		}
+	}
+
+	out, err := os.Create(path.Join(dir, "still.png"))
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	return png.Encode(out, g16)
+}
```

Comment:
It might be cleaner to have the snapshot related functionality in it's own file (snapshot.go or something?).